shared/utils/game_utils.py

```diff
@@ -284,6 +284,94 @@ def _check_single_king_and_throne(pieces: np.ndarray) -> bool:
     return True
 
 
+class Match(BaseModel):
+    """
+    Model representing a match in the Tablut game.
+
+    Attributes:
+        match_id (int): The unique identifier for the match.
+        white_player (AbstractPlayer): The player playing as white.
+        black_player (AbstractPlayer): The player playing as black.
+        turns (List[Turn]): The list of turns taken during the match.
+        states (List[State]): The list of states during the match.
+        actions (List[Action]): The list of actions taken during the match.
+        outcome (Turn): The outcome of the match.
+    """
+    from shared.utils import AbstractPlayer, State
+    match_id: int
+    white_player: AbstractPlayer
+    black_player: AbstractPlayer
+    turns: List[Turn]
+    states: List[State]
+    actions: List[Action]
+    outcome: Turn
+
+    def __str__(self) -> str:
```

Comment:
This is useless since pydantic.BaseModel can be json serialized with method .json(). Instead format it as a human readable string which lists the match id, the two player names, the state of the board and the action  for each turn and the winner